### PR TITLE
Fixed an issue where device options (e.g. -cart) were reported as unknown when they actually worked

### PR DESCRIPTION
**Pull request by npwoods**
This change also changes around how command line arguments are passed around; specifically I changed argc/argv to be std::vector<std::string>

Note this is not passed around 'const', the reason being that the command line processing will now "eat" the vector

### CODE REVIEW

File: src/frontend/mame/mame.cpp [hunk 1]
```diff
@@ -318,10 +318,18 @@ void emulator_info::display_ui_chooser(running_machine& machine)
...
+int emulator_info::start_frontend(emu_options &options, osd_interface &osd, std::vector<std::string> &args)
 {
 	cli_frontend frontend(options, osd);
-	return frontend.execute(argc, argv);
+	return frontend.execute(args);
+}
+
+int emulator_info::start_frontend(emu_options &options, osd_interface &osd, int argc, char *argv[])
+{
+	std::vector<std::string> args(argc);
```
Comment (cuavas) [inline src/frontend/mame/mame.cpp]: Why not use range-based construction and skip the loop?

    std::vector<std::string> args(argv, argv + argc);

File: src/lib/util/options.cpp [hunk 2]
```diff
@@ -487,16 +504,30 @@ const char *core_options::find_within_command_line(int argc, char **argv, const
...
-			[argv, i](const std::string &targ) { return targ == argv[i]; });
+			[&args, i](const std::string &targ) { return targ == args[i]; });
 		if (iter != targets.end())
-			return argv[i + 1];
+		{
+			// get the result
+			result = std::move(args[i + 1]);
+
+			// remove this arguments from the list (is there a standard
+			// way to do this?)
```
Comment (cuavas) [inline src/lib/util/options.cpp]: auto const pos = std::advance(args.begin(), i);
    args.erase(pos, std::advance(pos, 2));

**Comment by cuavas** [comment]
Anyone got an opinion on whether this is a sane way to do things?  @galibert maybe?

**Comment by galibert** [comment]
The more I think about it, the more I think it's sane.  I'd even have an
extra option to tell the final interpreted command line, which would help
answering questions about whether a given softlist does what's expected.

The file manager gives the option to switch easily between multiple images
present in the same softlist.  Do we want to have that option through the
command line (something like -flop1 image1,image2,image3)?

  OG.

**Comment by rb6502** [comment]
We have had the ability to select images from a softlist on the commandline since day 1 of softlists.

"apple2gs -flop3 sys601:flop1 -flop4 sys601:flop4" was valid before, for instance.

**Comment by mizapf** [comment]
This is not fixed yet, is it?

Error in command line:
Error: unknown option: -peb:slot3
Error: unknown option: -peb:slot8
Error: unknown option: -cart

**Comment by Tafoid** [comment]
Current git for me seems to be working fine:.
**[MINGW64] c:\mamegit\mame>mame64 ti99_4a -peb:slot3 32kmem -peb:slot8 speech -cart alpiner
Average speed: 100.00% (274 seconds)**

**Comment by mizapf** [comment]
Yes, it was obviously fixed shortly after my message. Works for me now.

**Comment by cuavas** [comment]
No it was fixed before your message.  You mustn't have pulled after it was merged.

**Comment by mizapf** [comment]
I was doing some bisecting around that time, and I noticed later that I did not do a reset, so maybe this was the problem. Anyway, it's working again, thanks.

**Pull request merged**